OpenRA.Mods.Common/Scripting/Properties/CombatProperties.cs

```diff
@@ -96,7 +96,7 @@ public void Attack(Actor targetActor, bool allowMove = true, bool forceAttack =
 			if (!targetActor.Info.HasTraitInfo<FrozenUnderFogInfo>() && !Self.Owner.CanTargetActor(targetActor))
 				Log.Write("lua", "{1} is not revealed for player {0}!", Self.Owner, targetActor);
 
-			attackBase.AttackTarget(target, true, allowMove, forceAttack);
+			attackBase.First(x => !x.IsTraitDisabled && x.HasAnyValidWeapons(target)).AttackTarget(target, true, allowMove, forceAttack);
```

Comment:
Please use FirstOrDefault and then throw a LuaException if it's null.  We don't want to leak cryptic C# exceptions into the scripting environment